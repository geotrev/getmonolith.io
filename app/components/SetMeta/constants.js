export const titleTemplate = placeholder => `Undernet â€“ ${placeholder}`
