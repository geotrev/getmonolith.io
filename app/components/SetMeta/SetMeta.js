import React from "react"
import PropTypes from "prop-types"
import { Helmet } from "react-helmet"

import { titleTemplate } from "./constants"

export default function SetMeta(props) {
  return (
    <Helmet titleTemplate={titleTemplate("%s")}>
      <title itemProp="name" lang="en">
        {props.title}
      </title>

      <meta name="description" content={props.description} />

      {!props.pageNotFound && <link rel="canonical" href={window.location.href} />}
      {props.pageNotFound && <meta name="prerender-status-code" content="404" />}
    </Helmet>
  )
}

SetMeta.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  pageNotFound: PropTypes.bool,
}

SetMeta.defaultProps = {
  title: "Page Not Found",
  description: "",
  pageNotFound: false,
}
