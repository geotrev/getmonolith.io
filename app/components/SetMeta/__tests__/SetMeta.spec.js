import React from "react"
import SetMeta from "../SetMeta"

describe("<SetMeta />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<SetMeta title="Test title" description="Test description" />)
    expect(wrapper).toMatchSnapshot()
  })
})
