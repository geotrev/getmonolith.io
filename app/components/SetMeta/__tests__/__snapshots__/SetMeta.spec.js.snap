// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetMeta /> renders 1`] = `
<SetMeta
  description="Test description"
  title="Test title"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="Undernet – %s"
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      link={
        Array [
          Object {
            "href": "http://localhost/",
            "rel": "canonical",
          },
        ]
      }
      meta={
        Array [
          Object {
            "content": "Test description",
            "name": "description",
          },
        ]
      }
      title="Test title"
      titleAttributes={
        Object {
          "itemprop": "name",
          "lang": "en",
        }
      }
      titleTemplate="Undernet – %s"
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        link={
          Array [
            Object {
              "href": "http://localhost/",
              "rel": "canonical",
            },
          ]
        }
        meta={
          Array [
            Object {
              "content": "Test description",
              "name": "description",
            },
          ]
        }
        title="Test title"
        titleAttributes={
          Object {
            "itemprop": "name",
            "lang": "en",
          }
        }
        titleTemplate="Undernet – %s"
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</SetMeta>
`;
